R/Driver.R

```diff
@@ -78,6 +78,28 @@ setMethod(
       uid = uid,
       pwd = pwd,
       .connection_string = .connection_string)
+
+    # perform the connection notification at the top level, to ensure that it's had
+    # a chance to get its external pointer connected, and so we can capture the
+    # expression that created it
+    if (!is.null(getOption("connectionObserver"))) {
+      addTaskCallback(function(expr, ...) {
+        tryCatch({
+          if (deparse(expr[[1]]) == "<-") {
```

Comment:
Probably something like `is.call(expr) && identical(expr[[1]], as.symbol("<-"))` would be a little more robust here.